packages/fxa-settings/src/lib/sensitive-data-client.ts

```diff
@@ -85,26 +88,12 @@ export namespace SensitiveData {
  * @class SensitiveDataClient
  */
 export class SensitiveDataClient {
-  /**
-   * @deprecated
-   */
-  private sensitiveData: { [key: string]: object } = {};
-
-  // TODO: Fast follow, use this pattern instead for simpler and better type safety.
-  public KeyStretchUpgradeData:
```

Comment:
Why would store things in a map when we could just store on an object?

Reply:
The map already existed before. I had moved the KeyStretchUpgradeData in a previous patch - this was the remaining bit. 

If you're asking for the general case - my goal was to create a pattern that allowed others to copy-paste without thinking about it too much. I don't have strong opinions besides consistency, so if we get everything using the same pattern we can move everything over to holding it in either one.

Reply:
I was one of the original reviewers on the sensitive data client. I wish I would have said something/caught this when it was created. When I added the KeyStrechUpgrade data field, it was my intent to go back in clean this up by switching over to using object fields.

As far as I can see, there really isn't a reason to have the map. We can use fields on the object directly. I'd recommend fixing this up. IMO, using a generic here is a misuse of generics.

Reply:
Dan and I spoke offline so we're documenting the conversation here for completeness.

The preference is to move towards having simple properties in the sensitive data client without the need for inserting them in a map first. The advantages are that it'll be easier to understand and follow, while also allowing for better IntelliSense support.

The aims of this patch were to incrementally add better types to this work. Since we want to move to this future, I'm going to keep the `KeyStretchUpgradeData` as best we can for now with a follow-up issue to move the sensitive data client to use properties in it.

I've filed FXA-10929 to track this future work. Thanks for the discussion!